### PR TITLE
New Nginx Tracing Module

### PR DESCRIPTION
**Pull request by dgoffredo**
### What does this PR do?
This revision adds onboarding documentation for the new nginx module.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 1]
```diff
@@ -298,10 +298,66 @@ The available [environment variables][3] depend on the version of the C++ tracer
...
+NGINX_IMAGE_TAG=1.23.1
+RELEASE_TAG=$(get_latest_release DataDog/nginx-datadog)
+tarball="$NGINX_IMAGE_TAG-ngx_http_datadog_module.so.tgz"
+wget "https://github.com/DataDog/nginx-datadog/releases/download/$RELEASE_TAG/$tarball"
+tar -xzf "$tarball" -C /usr/lib/nginx/modules
+rm "$tarball"
+```
+
+### Nginx configuration with Datadog module
+In the topmost section of the nginx configuration, load the Datadog module.
```
Comment (hestonhoffman) [inline content/en/tracing/trace_collection/proxy_setup/_index.md:333]:
```suggestion
In the topmost section of the Nginx configuration, load the Datadog module.
```

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 2]
```diff
@@ -678,6 +741,12 @@ If using Kubernetes 1.18+, `appProtocol: tcp` can be added to the port specifica
 [10]: /getting_started/tagging/unified_service_tagging/?tab=kubernetes#configuration-1
 [11]: /tracing/setup/cpp/#environment-variables
 [12]: https://istio.io/docs/ops/configuration/traffic-management/protocol-selection/#manual-protocol-selection
+[13]: https://hub.docker.com/_/nginx/tags
+[14]: https://github.com/DataDog/nginx-datadog/releases/latest
+[15]: https://hub.docker.com/layers/nginx/library/nginx/1.23.1/images/sha256-f26fbadb0acab4a21ecb4e337a326907e61fbec36c9a9b52e725669d99ed1261?context=explore
+[16]: https://hub.docker.com/layers/nginx/library/nginx/1.23.1-alpine/images/sha256-2959a35e1b1e61e2419c01e0e457f75497e02d039360a658b66ff2d4caab19c4?context=explore
+[17]: https://github.com/DataDog/nginx-datadog/blob/master/doc/API.md
+[18]: https://github.com/DataDog/nginx-datadog/blob/master/doc/API.md#datadog
```
[the remark this answers is not in the document]
Comment (dgoffredo) [reply content/en/tracing/trace_collection/proxy_setup/_index.md:749]: Ah, that explains it.  Thanks.

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 3]
```diff
@@ -298,10 +298,66 @@ The available [environment variables][3] depend on the version of the C++ tracer
...
+The default configuration connects to a local Datadog Agent and produces traces
+for all Nginx locations. Specify custom configuration in a `datadog` JSON block
+within the `http` section of the nginx configuration.
+
+```nginx
+load_module modules/ngx_http_datadog_module.so;
+
+http {
+  datadog {
+    "service": "usage-internal-nginx",
```
Comment (yoannrobin) [inline content/en/tracing/trace_collection/proxy_setup/_index.md:348]: Is service part of the documentation? (https://github.com/DataDog/nginx-datadog/blob/master/doc/API.md#datadog)

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 4]
```diff
@@ -298,10 +298,66 @@ The available [environment variables][3] depend on the version of the C++ tracer
...
+In the topmost section of the nginx configuration, load the Datadog module.
+
+```nginx
+load_module modules/ngx_http_datadog_module.so;
+```
+
+The default configuration connects to a local Datadog Agent and produces traces
+for all Nginx locations. Specify custom configuration in a `datadog` JSON block
+within the `http` section of the nginx configuration.
+
```
Comment (yoannrobin) [inline content/en/tracing/trace_collection/proxy_setup/_index.md:342]:
```suggestion
As an example, to set a service name to `usage-internal-nginx` and sample 10% of your traces:
```

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 5]
```diff
@@ -398,7 +461,7 @@ To control the volume of Nginx traces that are sent to Datadog, specify a sampli
 }
 ```
 
-If no value is specified, the [Datadog Agent calculated sampling rates][7] (10 traces per second per Agent) are applied.
+If no sample rate is specified, the [Datadog Agent calculated sampling rates][7] (10 traces per second per Agent) are applied.
```
Comment (yoannrobin) [inline content/en/tracing/trace_collection/proxy_setup/_index.md:464]: Does it also apply to Nginx module? As this section falls under the Nginx with opentracing module I would assume it is only relevant for this case

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 6]
```diff
@@ -298,10 +298,66 @@ The available [environment variables][3] depend on the version of the C++ tracer
...
+    "service": "usage-internal-nginx",
+    "sample_rate": 0.1
+  }
+}
+```
+
+For information about fields supported by the `datadog` directive and about
+other configuration directives supported by the module, see the [API
+documentation][17].
+
```
Comment (yoannrobin) [inline content/en/tracing/trace_collection/proxy_setup/_index.md:357]: Should we add the evaluation order between directive and environment variables?

File: content/en/tracing/trace_collection/proxy_setup/_index.md (hunk 3 above)
Comment (dgoffredo) [reply content/en/tracing/trace_collection/proxy_setup/_index.md:348]: yes

File: content/en/tracing/trace_collection/proxy_setup/_index.md (hunk 5 above)
Comment (dgoffredo) [reply content/en/tracing/trace_collection/proxy_setup/_index.md:464]: `### Nginx Sampling` is an `h3` section, the same as the OpenTracing and Datadog module sections.  So, there is one section about getting the Datadog module, another about getting the OpenTracing module, and then a third about configuring sampling for them.

This is possibly confusing.  What might be better?

Comment (dgoffredo) [reply content/en/tracing/trace_collection/proxy_setup/_index.md:464]: I am wrong.  Will fix.

File: content/en/tracing/trace_collection/proxy_setup/_index.md (hunk 6 above)
Comment (dgoffredo) [reply content/en/tracing/trace_collection/proxy_setup/_index.md:357]: That's covered in the documentation of the environment variables, which is linked in the API documentation.

File: content/en/tracing/trace_collection/proxy_setup/_index.md [hunk 7]
```diff
@@ -298,10 +298,69 @@ The available [environment variables][3] depend on the version of the C++ tracer
...
-The instructions below use Nginx from the official [Linux repositories][1] and pre-built binaries for the plugins.
+Datadog APM supports Nginx in multiple configurations:
+- Nginx operated as a proxy with tracing provided by the new Datadog module.
+- Nginx operated as a proxy with tracing provided by the OpenTracing module.
+- Nginx as an Ingress Controller for Kubernetes.
 
-## Nginx open source
+## Nginx with Datadog module
+Datadog provides an Nginx module for distributed tracing. This module is 
+currently in beta.
```
Comment (yoannrobin) [inline content/en/tracing/trace_collection/proxy_setup/_index.md:308]:
```suggestion
<div class="alert alert-warning">
Datadog Nginx module is currently in beta. <a href="https://docs.datadoghq.com/help/">Reach out to support</a> if you have any feedback.
</div>
```

**Review by yoannrobin — APPROVED** [review]

**Comment by dgoffredo** [comment]
This is ready for editorial review.

Thanks for the comments everyone so far.

**Comment by dgoffredo** [comment]
Hey @hestonhoffman, are you a code owner who can approve and merge this PR, or should I @ someone else?

**Comment by hestonhoffman** [comment]
@dgoffredo Yes! I'll take a quick look after lunch and approve/merge

**Review by hestonhoffman — APPROVED** [review]

**Comment by dgoffredo** [comment]
Thanks!

**Pull request merged**